X-CASH-official/desktop-wallet
# Download xcash binaries in development mode if they do not exist

# Description

Download xcash wallet 2.0 binaries while configuring user working directory. At the moment it's not possible to start the wallet in development mode because it expects to have `xcash-wallet-rpc` in the `$HOME/xcash-official` folder.

In particular, the process is:
1. Check if the user's working directory already exists. If it exists => do nothing;
2. Download binaries depending on user's OS;
3. Unzip it;
4. Move `xcash-wallet-rpc` to the user's working directory;
5. Grant privileges for *NIX systems.

Also,
- MacOS build instructions are fixed;
- Fixed a few warnings and wrongly resolved promises;
- Code style formatting is applied when changes are introduced;

## Type of change

Please delete options that are not relevant.


# How Has This Been Tested?

* Take a fresh system (Linux/MacOS/Windows);
* Clone the repo;
* Install dependencies;
* Build the wallet from source for your OS;
* Run the wallet. 

`xcash-wallet-rpc`is expected to be placed into `$HOME/xcash-official` with necessary privileges granted.
  

# Checklist:
